code/datums/components/virtual_entity.dm

```diff
@@ -0,0 +1,19 @@
+/// Handles all special considerations for "virtual entities" such as bitrunning ghost roles or digital anomaly antagonists.
+/datum/component/virtual_entity
+	///The cooldown for balloon alerts, so the player isn't spammed while trying to enter a restricted area.
+	COOLDOWN_DECLARE(OOB_cooldown)
+
+/datum/component/virtual_entity/Initialize()
+	. = ..()
+	RegisterSignal(parent, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(on_parent_pre_move))
+
+/datum/component/virtual_entity/proc/on_parent_pre_move(atom/movable/source, atom/new_location)
+	var/area/location_area = get_area(new_location)
```

Comment:
```suggestion
/datum/component/virtual_entity/proc/on_parent_pre_move(atom/movable/source, atom/new_location)
	SIGNAL_HANDLER
	var/area/location_area = get_area(new_location)
```

Reply:
also write an autodoc comment for this pls- just what its doing/when it fires